substrate/primitives/core/src/bls.rs

```diff
@@ -227,6 +241,49 @@ impl<T: BlsBound> TraitPair for Pair<T> {
 	}
 }
 
+impl<T: BlsBound> ProofOfPossessionGenerator for Pair<T> {
+	#[cfg(feature = "full_crypto")]
+	fn generate_proof_of_possession(&mut self) -> Self::Signature {
+		let r: [u8; SIGNATURE_SERIALIZED_SIZE] = <Keypair<T> as BlsProofOfPossessionGenerator<
+			T,
+			Sha256,
```

Comment:
Choice of Sha256 as the hasher is in the spec?

Reply:
IETF standards mandates SHA256 for hash to curve (https://www.rfc-editor.org/rfc/rfc9380.html#name-bls12-381-g1).  So may as well use it for other part of the system, so if someone wants to verify proof of possession they need to implement the hash to curve and so it needs to implement SHA256 then may as well we use it for Fiat-Shamir as well. If SHA256 is broken then the hash to curve anyway.